### PR TITLE
Adds Professor Page BackEnd

### PR DESCRIPTION
**Pull request by hailed748**
Adds new servlets which will process data for all terms taught by a specific professor Adds: ProfTermHold.java, ProfessorData.java, ProfessorDataTest.java. Also adds new professor.html template page

### CODE REVIEW

File: src/main/webapp/professor.html [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      name="google-signin-client_id"
+      content="12523748853-6h3fnh1kppht9s895ist2b5222ifh2u7.apps.googleusercontent.com"
+    />
+    <link rel="stylesheet" href="course-style.css" />
+  </head>
+  <body>
+    <nav class="navbar navbar-dark bg-dark">
+      <span class="navbar-brand mb-0 h1">App Name</span>
+    </nav>
+
```
Comment (ninaprabhu) [inline src/main/webapp/professor.html:35]: are you supposed to have a space here?

File: src/test/java/com/google/sps/servlets/ProfessorDataTest.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    List<Object> expectedCommentsListA = new ArrayList(Arrays.asList("Great"));
+
+    List<Object> expectedHoursListB = new ArrayList(Arrays.asList((long) 11));
+    List<Object> expectedDifficultyListB = new ArrayList(Arrays.asList((long) 6));
+    List<Object> expectedPerceptionListB = new ArrayList(Arrays.asList((double) 0.79));
+    List<Object> expectedCommentsListB = new ArrayList(Arrays.asList("Good"));
+
+    createRequest(
+        /* requestServelt */ request,
+        /* schoolName */ "MIT",
```
Comment (ninaprabhu) [inline src/test/java/com/google/sps/servlets/ProfessorDataTest.java:84]: Not super relevant now, but we'll probably need to lowercase this later as the email format is "mit.edu" (or do some kind of display name vs. email format kinda thing)

Comment (ninaprabhu) [reply src/test/java/com/google/sps/servlets/ProfessorDataTest.java:84]: ^But something we should start thinking about, so implement if possible

File: src/test/java/com/google/sps/servlets/ProfessorDataTest.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+
+    List<Object> expectedHoursListA = new ArrayList(Arrays.asList((long) 12));
+    List<Object> expectedDifficultyListA = new ArrayList(Arrays.asList((long) 7));
+    List<Object> expectedPerceptionListA = new ArrayList(Arrays.asList((double) 0.8));
+    List<Object> expectedCommentsListA = new ArrayList(Arrays.asList("Great"));
+
+    List<Object> expectedHoursListB = new ArrayList(Arrays.asList((long) 11));
+    List<Object> expectedDifficultyListB = new ArrayList(Arrays.asList((long) 6));
+    List<Object> expectedPerceptionListB = new ArrayList(Arrays.asList((double) 0.79));
```
Comment (lqueipom) [inline src/test/java/com/google/sps/servlets/ProfessorDataTest.java:79]: why is this one not just a long?

**Comment by LukeGMTravis** [comment]
Starting review

File: src/main/java/com/google/sps/data/ProfDataHolder.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProfDataHolder {
```
Comment (LukeGMTravis) [inline src/main/java/com/google/sps/data/ProfDataHolder.java:6]: Should probably note somewhere here (at least a comment) that this is per term. Maybe renaming the class to something like ProfDataForTerm, etc would be better though...

File: src/main/webapp/professor.html [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      <div class="flex-child-p" id="term-perception-comp"></div>
+    </div>
+    <div class="plr-20 container-fluid">
+      <div class="my-3 p-3 bg-white" id="prof-comments">
+        <h1 class="border-bottom border-gray pb-5 mb-0 text-center">
+          Professor Comments
+        </h1>
+      </div>
+    </div>
+    <script src="https://cdn.jsdelivr.net/npm/jstat@1.9.3/dist/jstat.min.js"></script>
```
Comment (LukeGMTravis) [inline src/main/webapp/professor.html:58]: Can all script tags be placed together for better organization?

File: src/test/java/com/google/sps/servlets/ProfessorDataTest.java [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+    schoolData = new AddSchoolData();
+    professorData = new ProfessorData();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
```
Comment (LukeGMTravis) [inline src/test/java/com/google/sps/servlets/ProfessorDataTest.java:64]: This should be "requestA" for consistency

File: src/test/java/com/google/sps/servlets/ProfessorDataTest.java [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+        /* requestServelt */ requestB,
+        /* schoolName */ "mit",
+        /* courseName */ "6.006",
+        /* termName */ "Fall 2020",
+        /* units */ "12",
+        /* profName */ "Jason Ku");
+
+    schoolData.addSchoolData(db, request);
+    schoolData.addSchoolData(db, requestB);
+    Key parent = findQueryMatch(db, "Term", "term", "Spring 2020").get(0).getKey();
```
Comment (LukeGMTravis) [inline src/test/java/com/google/sps/servlets/ProfessorDataTest.java:99]: Should be "parent1" for consistency

Comment (LukeGMTravis) [inline src/test/java/com/google/sps/servlets/ProfessorDataTest.java:99]: Just a thought, this entity (or at least the key) feels rather like something that "addSchoolData" should be returning

Comment (hailed748) [reply src/test/java/com/google/sps/servlets/ProfessorDataTest.java:99]: I will look into this. Let me get all the professor related PR's in then I will go back and refactor this

Comment (LukeGMTravis) [reply src/test/java/com/google/sps/servlets/ProfessorDataTest.java:99]: Good to add a TODO comment in that case so you can track this

**Pull request merged**